### ISSUE TITLE
Problem to run project

### ISSUE THREAD

**Issue by Rafal221373** [RevKarl/MLX90640-python#1]
I have question about how to run this python code. I create 2 python file one with MLX90640 and second main. When I started main.py I receive name self is not defined in line self.address=address. I thought you miss self value in __init__(self,address), but after do this I have more errors. I add smbbus2 library and also have raspberry pi 

main.py  :
import MLX90640
sensor = MLX90640.MLX90640() 
sensor.getCompensatedPixData(1,1)

**Comment by vvkuryshev** [comment]
I agree with the remark above. After adding init (**self**, address), the following errors are displayed:

  File "main.py", line 3, in <module>
    print(sensor.getCompensatedPixData(1,1))
  File "/home/pi/projects/mlx90640/MLX90640.py", line 224, in getCompensatedPixData
    pixOs = self.getPixData(i,j)
  File "/home/pi/projects/mlx90640/MLX90640.py", line 175, in getPixData
    OCCRow = (self.getRegf(rowAdd) & rowMask)>>(4*((i-1)%4))
  File "/home/pi/projects/mlx90640/MLX90640.py", line 36, in getRegf
    write = i2c_msg.write(self.addr,[reg>>8,reg&0xFF])
TypeError: unsupported operand type(s) for >>: 'float' and 'int'

**Comment by padchin** [comment]
200 KtaRC = (self.getRegf(KtaAvAddr) & KtaAvMask) >> 8 * rowOdd
201        if KtaRC > 127:
202            KtaRC = KtaAvRC - 256

KtaAvRC - unresolved reference

**Comment by herrtschaefer** [comment]
Had the same issues, how did this work for you? well, i modified it slightly to cast the float to int and then it worked "kindof". And i also don't know if the data i'm getting is actually valid.
I'm pretty new in python but i like your project so i want to ask you if you would update the library so that it works.

**Comment by padchin** [comment]
Hi. We made the chip transfer the correct data in another way.

**Comment by herrtschaefer** [comment]
> Hi. We made the chip transfer the correct data in another way.

So in what way are you transfering the data? I'm getting pretty much unusable data out of it. To show how i modyfied the code i uploaded my version, maybe someone can spot an error? I'd much appreciate a little help.
[MLXModified.txt](https://github.com/RevKarl/MLX90640-python/files/2898154/MLXModified.txt)
Thanks in advance.

**Comment by padchin** [comment]
https://m.habr.com/ru/post/435946/

In this post there is an example of working code. The article is in Russian, but the python speaks for itself.

**Comment by herrtschaefer** [comment]
So you can't use the standard  sensor like the sparkfun version with this code?

**Comment by padchin** [comment]
Unfortunately, I can not say anything about this.

**Comment by vvkuryshev** [comment]
Yes, the code works with TI micro controller on GY-MLX90640 board. You can read about it here (in English): https://habr.com/en/post/441050/. Or in Russian at the link three posts ago.
